pkg/retester/s3BackOffCache.go

```diff
@@ -0,0 +1,108 @@
+package retester
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"io"
+	"k8s.io/test-infra/prow/tide"
+	"time"
+)
+
+const (
+	retesterBucket = "prow-retester"
+)
+
+type s3BackOffCache struct {
+	cache          map[string]*pullRequest
+	file           string
+	cacheRecordAge time.Duration
+	logger         *logrus.Entry
+
+	awsSession *session.Session
+}
+
+func (b *s3BackOffCache) load() error {
+	return b.loadFromAwsNow(time.Now())
+}
+
+// loadFromAwsNow gets the backoff cache file from AWS S3 bucket and marshals its content into the s3BackOffCache
+func (b *s3BackOffCache) loadFromAwsNow(now time.Time) error {
+	if b.file == "" {
+		return nil
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String("us-east-1"),
+	})
+	if err != nil {
+		return fmt.Errorf("couldn't create new aws session")
+	}
+	b.awsSession = sess
+
+	_, err = sess.Config.Credentials.Get()
+	if err != nil {
+		return fmt.Errorf("credentials for aws not found")
+	}
+
+	svc := s3.New(b.awsSession)
+	result, err := svc.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(retesterBucket),
+		Key:    aws.String(b.file),
+	})
+	if err != nil {
+		return fmt.Errorf("couldn't get %s file from aws s3 bucket %s: %w", b.file, retesterBucket, err)
+	}
+
+	content, err := io.ReadAll(result.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read file %s: %w", b.file, err)
+	}
+
+	cache := map[string]*pullRequest{}
```

Comment:
We could make a shared function (with `fileBackOffCache `) for this too.

loadAndDelete(content byte[], cache map[string]*pullRequest)

Reply:
You mean **one** func for unmarshalling the []byte content into `cache` and also deleting the old records?